contracts/generic/AugmintToken.sol

```diff
@@ -143,15 +144,21 @@ contract AugmintToken is AugmintTokenInterface {
     }
 
     function _transfer(address from, address to, uint256 amount, string narrative) private {
-        require(to != 0x0);
+        require(to != 0x0, "to must be set");
+
         uint fee = feeAccount.calculateTransferFee(from, to, amount);
+        uint transferAmount = amount.add(fee);
+
+        // to emit proper reason instead of failing on from.sub()
+        require(balances[from] >= transferAmount, "balance must be >= amount + transfer fee");
+
         if (fee > 0) {
             balances[feeAccount] = balances[feeAccount].add(fee);
-            balances[from] = balances[from].sub(amount).sub(fee);
```

Comment:
sure about this change? a semantic change, not much to do with solc update...

Reply:
got it. looks okay, though I would rename `transferAmount` (since it's not the amount transferred) to something like `totalCost`, `totalExpense` or such...